### PR TITLE
feat: ✨ map support

### PR DESCRIPTION
**Pull request by alvaro-saavedra-sumup**
## Describe your changes

Add support for maps

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests (if possible)

### CODE REVIEW

File: src/visitor.ts [hunk 1]
```diff
@@ -164,12 +186,64 @@ export class CelVisitor
...
+    return [key, value]
+  }
+
   macrosExpression(ctx: MacrosExpressionCstChildren): unknown {
     const macrosIdentifier = ctx.MacrosIdentifier[0]
     // eslint-disable-next-line sonarjs/no-small-switch
     switch (macrosIdentifier.image) {
-      case 'size':
-        return ctx.arg ? this.visit(ctx.arg).length : 0
+      case 'size': // todo type it
```
Comment (ChromeGG) [inline src/visitor.ts:232]: Probably we can type it in the similar way as reserved keywords are typed. But it's not a blocker, can be done in a separate PR 😄

File: src/visitor.ts [hunk 2]
```diff
@@ -164,12 +186,64 @@ export class CelVisitor
...
+        return ctx.arg ? size(this.visit(ctx.arg)) : 0
```
Comment (ChromeGG) [inline src/visitor.ts:233]: NIT: our `size()` returns `0`, the original one throws an error. Also not a blocker

File: src/spec/maps.spec.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+  describe('addition', () => {
+    it('should add to values accessed by index', () => {
+      const expr = '{"a": 1, "b": 2}["a"] + {"a": 1, "b": 2}["b"]'
+
+      const result = evaluate(expr)
+
+      expect(result).toStrictEqual(3)
+    })
+  })
```
Comment (ChromeGG) [inline src/spec/maps.spec.ts:113]: I am not sure if we need this test. It's just a combination of index access and addition, nothing specific to maps 🤔

File: src/helper.ts [hunk 4]
```diff
@@ -22,6 +22,7 @@ import {
   IdentifierDotExpressionCstNode,
   IndexExpressionCstNode,
 } from './cst-definitions.js'
+import { equals } from 'ramda'
```
Comment (ChromeGG) [inline src/helper.ts:25]: `assert` was node specific, I replaced it to ramda's `equals`. This solved the issue and simplified code

**Review by ChromeGG — APPROVED** [review]
Amazing job, man! Thank you for you contribution! <3

We have to remember about two small things:
1. Always run regenerate `cst-definitios.ts`,
2. Format code using default prettier rules.

ideally, it should be part of the CI.

Optionally, you can fix the PR issues. PR has my approval, so it will be merged and released in the next 3 days 😄

PS What is your motivation for supporting this open source? Are you using it in your company? 😄

File: src/helper.ts (hunk 4 above)
Comment (alvaro-saavedra-sumup) [reply src/helper.ts:25]: perfect!!!

**Comment by alvaro-saavedra-sumup** [comment]
> PS What is your motivation for supporting this open source? Are you using it in your company? 😄

Not "jet", I did a POC for an internal usage software, and I was missing some features, but now is just for fun.

**Pull request merged**